pyradiosky/skymodel.py

```diff
@@ -1260,6 +1283,258 @@ def coherency_calc(self, deprecated_location=None):
+                    )
+                )
+
+        if (
+            this.extended_model_group is not None
+            and other.extended_model_group is not None
+        ):
+            this.extended_model_group = np.concatenate(
+                (this.extended_model_group, other.extended_model_group)
+            )
+        elif this.extended_model_group is not None:
+            warnings.warn(
+                "This object has extended_model_group values, other object does not. "
+                "Filling missing values with zeros."
+            )
+            this.extended_model_group = np.concatenate(
+                (
+                    this.extended_model_group,
+                    np.full(
+                        other.Ncomponents, "", dtype=this.extended_model_group.dtype
+                    ),
+                )
+            )
+        elif other.extended_model_group is not None:
+            warnings.warn(
+                "This object does not have extended_model_group values, other object does. "
+                "Filling missing values with zeros."
+            )
+            this.extended_model_group = np.concatenate(
+                (
+                    np.full(
+                        this.Ncomponents, "", dtype=other.extended_model_group.dtype
+                    ),
+                    other.extended_model_group,
+                )
+            )
+
+        if this.beam_amp is not None and other.beam_amp is not None:
+            this.beam_amp = np.concatenate((this.beam_amp, other.beam_amp), axis=2)
+        elif this.beam_amp is not None:
+            warnings.warn(
+                "This object has beam_amp values, other object does not. "
+                "Filling missing values with NaNs."
+            )
+            this.beam_amp = np.concatenate(
+                (
+                    this.beam_amp,
+                    np.full(
+                        (4, other.Nfreqs, other.Ncomponents),
+                        None,
+                        dtype=this.beam_amp.dtype,
+                    ),
+                ),
+                axis=2,
+            )
+        elif other.beam_amp is not None:
+            warnings.warn(
+                "This object does not have beam_amp values, other object does. "
+                "Filling missing values with NaNs."
+            )
+            this.beam_amp = np.concatenate(
+                (
+                    np.full(
+                        (4, other.Nfreqs, other.Ncomponents),
+                        None,
+                        dtype=other.beam_amp.dtype,
+                    ),
+                    other.beam_amp,
+                ),
+                axis=2,
+            )
+        this.Ncomponents = this.Ncomponents + other.Ncomponents
+
+        history_update_string = (
+            " Combined skymodels along the component axis using pyradiosky."
+        )
+        this.history += history_update_string
```

Comment:
Does this put the `history_update_string` in between the `this` history and the `other` history?  Is that where we want it?

Reply:
oops, that's a mistake, thanks for catching it. I'll need to improve the testing to catch this.

Reply:
Although I now see that this matches what we do in pyuvdata. So I think there's a question about what we want in both places.

Reply:
I've changed this as discussed in the telecon and I'm putting together a matching PR on pyuvdata.